src/Periphery/TokenWrapper.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.17;
+
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { TransferrableOwnership } from "../Helpers/TransferrableOwnership.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// External wrapper interface
+interface IWrapper {
+    function deposit() external payable;
+    function withdraw(uint wad) external;
+}
+
+/// @title TokenWrapper
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for wrapping and unwrapping tokens
+/// @custom:version 1.0.0
+contract TokenWrapper is TransferrableOwnership {
+    uint256 private constant MAX_INT = 2**256 - 1;
+    address public wrappedToken;
+
+    /// Errors ///
+    error WithdrawFailure();
+
+    /// Events ///
+    event Wrapped(uint256 amount);
+    event Unwrapped(uint256 amount);
+
+    /// Constructor ///
+    // solhint-disable-next-line no-empty-blocks
+    constructor(address _owner, address _wrappedToken) TransferrableOwnership(_owner) {
+        wrappedToken = _wrappedToken;
+        IERC20(wrappedToken).approve(address(this), MAX_INT);
+    }
+
+    /// External Methods ///
+
+    /// @notice Wraps the native token
+    function deposit(
+    ) external payable {
+        IWrapper(wrappedToken).deposit{value: msg.value}();
+        IERC20(wrappedToken).transfer(msg.sender, msg.value);
+        emit Wrapped(msg.value);
```

Comment:
Do we need events here? Our contracts already expose assetSwapped events for each executed swap. I don't think the event here is needed. It would save some gas (also `Unwrapped(data)` event)

Reply:
Sure